### ISSUE TITLE
What is a math expression?

### ISSUE THREAD

**Issue by 9999years** [nushell/nushell#12957]
### Question

What is a "math expression"? What's the grammar for a math expression and where do they occur in the grammar?

### Additional context and details

This code works fine:

```nu
> let $name = ("Rebecca")
```

This code produces an error, in a place I wouldn't expect the grammar to be whitespace-sensitive:

```nu
> let $name =("Rebecca")
Error: nu::parser::expected_keyword

  × Expected keyword.
   ╭─[entry #89:1:11]
 1 │ let $name =("foo")
   ·           ────┬───
   ·               ╰── expected =
   ╰────
```

(The language server annotates this as an incomplete math expression.)

Moreover, in which places _is_ the grammar whitespace sensitive?

**Comment by 9999years** [comment]
Yes, but *where* is the grammar whitespace-sensitive? E.g. lists aren't: `[1,2,3]` works just as well as `[1, 2, 3]`.

**Comment by fdncred** [comment]
There are many places. `1+2` won't work. `def main[]{}` won't work. I'm not sure there is a list of where it is though.

**Comment by WindSoilder** [comment]
Closing the issue because the question is answered.

**Comment by 9999years** [comment]
I'm sorry, but the question wasn't answered: there is not a list of places where the grammar is whitespace sensitive, much less a full listing of the grammar itself.

Also, I should have made this clearer in the original issue, but this is a documentation ticket: the grammar should be _documented,_ it's not enough for it to be listed in a GitHub issue somewhere.
